tgui/packages/tgui/interfaces/VendingConsole.tsx

```diff
@@ -0,0 +1,132 @@
+import { map } from 'common/collections';
+import { useBackend, useLocalState } from '../backend';
+import { Box, ProgressBar } from '../components';
+import { Window } from '../layouts';
+
+import {
+  NanoMap,
+  NanoMapMarkerIcon,
+  NanoMapStaticPayload,
+  NanoMapTrackData,
+} from '../components/NanoMap';
+
+const pickColor = (machine: VendingObject) => {
+  let color = 'green';
+  if (machine.status === 2) {
+    color = 'grey';
+  } else if (machine.status === 3) {
+    color = 'red';
+  } else if (machine.load < 75) {
+    color = 'yellow';
+  } else if (machine.load < 50) {
+    color = 'orange';
+  } else if (machine.load < 25) {
+    color = 'red';
+  }
+
+  return { color };
+};
+
+const pauseEvent = (e: MouseEvent) => {
+  if (e.stopPropagation) {
+    e.stopPropagation();
+  }
+  if (e.preventDefault) {
+    e.preventDefault();
+  }
+  e.cancelBubble = true;
+  e.returnValue = false;
+  return false;
+};
+
+const tooltipForMachine = (machine: VendingObject) => (
+  <Box textAlign="center">
+    {<b>{machine.name}</b>}:{' '}
+    {
+      <i>
+        {machine.status === 3
+          ? 'Сломан'
+          : machine.status === 2
+            ? 'Обесточен'
+            : 'Работает'}
+      </i>
+    }
+    {machine.status === 1 && (
+      <ProgressBar
+        ranges={{
+          green: [75, Infinity],
+          yellow: [50, 75],
+          orange: [25, 50],
+          red: [-Infinity, 25],
+        }}
+        minValue={0}
+        maxValue={100}
+        value={machine.load}
+      />
+    )}
+  </Box>
+);
+
+type Data = {
+  vending_machines: VendingObject[];
```

Comment:
сори лучше либо снейк кейс либо кэмэл, везде в тгуи кэмэл используется, так что...

```suggestion
  vendingMachines: VendingObject[];
```